core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -596,7 +597,16 @@ protected void onModified() throws IOException {
      * TCP agent port.
      * 0 for random, -1 to disable.
      */
-    private int slaveAgentPort = SystemProperties.getInteger(Jenkins.class.getName()+".slaveAgentPort",0);
+    private int slaveAgentPort = getSlaveAgentPortInitialValue(0);
+
+    private static int getSlaveAgentPortInitialValue(int def) {
+        return SystemProperties.getInteger(Jenkins.class.getName()+".slaveAgentPort", def);
+    }
+
+    /**
+     * If -Djenkins.model.Jenkins.slaveAgentPort is defined, enforce it on every start instead of only the first one.
+     */
+    private static final boolean SLAVE_AGENT_PORT_ENFORCE = SystemProperties.getBoolean(Jenkins.class.getName()+".slaveAgentPortEnforce", false);
```

Comment:
🐛 From what I see it does not really enforce it since a Jenkins admin can actually change it. IMHO configuration flow needs to be adjusted:
* Values from the form submissions should be ignored
* Ideally the field should be readonly when this flag is set
* It should be possible forcing the disabled state (paranoid mode)